fairscale/experimental/wgit/sha1_store.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+import json
+import os
+from pathlib import Path
+import sys
+import typing
+
+from fairscale.experimental.wgit.utils import ExitCode
+
+
+class SHA1_store:
+    """
+    Planned Features:
+        1. def init
+        2. def add <file or data> -> SHA1
+        3. def remove (SHA1)
+        4. def add_ref(children_SHA1, parent_SHA1)
+        5. def read(SHA1): ->
+        6. def lookup(SHA1): -> file path to the data. NotFound Exception if not found.
+    """
+
+    def __init__(self) -> None:
+        try:
+            sha1_store_path = Path.cwd().joinpath(".wgit", "sha1_store")
+            if not sha1_store_path.exists():
+                Path.mkdir(sha1_store_path, parents=False, exist_ok=False)
+        except FileExistsError as error:
+            sys.stderr.write(f"An exception occured while creating Sha1_store: {repr(error)}\n")
+            sys.exit(ExitCode.FILE_EXISTS_ERROR)
+
+    def add_ref(self, current_sha1_hash: str) -> None:
```

Comment:
Just to make sure we don't have any misunderstanding, the sha1_store.add() function does more than just checksuming, right? It also copy the file or store the data into a destination file inside sha1_store's internal dir so that it can be read back in the future, right?